jdaviz/app.py

```diff
@@ -887,6 +848,190 @@ def _get_all_subregions(mask, spec_axis_data):
 
         return regions
 
+    def get_subsets(self, subset_name=None, spectral_only=False,
+                    spatial_only=False, astropy_region=False):
+        """
+        Returns all branches of glue subset tree in the form that subset plugin can recognize.
+
+        Parameters
+        ----------
+        subset_name : str
+            The subset name.
+        spectral_only : bool
+            Return only spectral subsets.
+        spatial_only : bool
+            Return only spatial subsets.
+        astropy_region : bool
```

Comment:
This keyword is misleading. I think our Subsets actually return objects from completely different packages depending on the type.

1. Spatial returns `regions` object. This package is a.k.a. "astropy regions" to differentiate it from `pyregions`.
2. Spectral returns `specutils` object, which does not talk to `regions` at all.

Reply:
`astropy_region` is really just a bool to decide if we return the composite subsets as `{"name": Circle, "glue_state": ROI, "region": <RegionObject>}` or just `<RegionObject>`, which is how `get_subsets_from_viewer` returns the information for a simple ROI.

Reply:
I propose we rename it to something less confusing because we are not strictly talking about the `regions` _package_ here.

Reply:
`object_only`? I would avoid `region` since people can confuse it with the `regions` package, or that might just be me...